### PR TITLE
Add extra file to avoid "Failed to open file" error

### PR DESCRIPTION
**Pull request by kylazhang**
Signed-off-by: Kyla Zhang <weizhan@redhat.com>

### CODE REVIEW

File: virttest/utils_test/libvirt.py [hunk 1]
```diff
@@ -2710,7 +2711,9 @@ def attach_additional_device(vm_name, targetdev, disk_path, params, config=True)
     params['target_dev'] = targetdev
 
     # Create a file of device
-    xmlfile = create_disk_xml(params)
+    xmlfile = os.path.join(data_dir.get_tmp_dir(), "xml_file")
+    tmpfile = create_disk_xml(params)
+    shutil.copyfile(tmpfile, xmlfile)
```
Comment (dzhengfy) [inline virttest/utils_test/libvirt.py:2716]: This problem might be caused by create_disk_xml(). Could you test the similar fix in https://github.com/avocado-framework/avocado-vt/pull/2988 to see if it solves your problem?

Comment (kylazhang) [reply virttest/utils_test/libvirt.py:2716]: @dzhengfy Just checking the patch for #2988 but I don't think change the return value is a good way as it is widely used.

Comment (chunfuwen) [reply virttest/utils_test/libvirt.py:2716]: From running test result on latest release, it sound likes there are a few places to throw this error.
We may need investigate root clause and  check whether there is uniform place to fix that.

**Review by chunfuwen — APPROVED** [review]
let's approve and have a try.

File: virttest/utils_test/libvirt.py (hunk 1 above)
Comment (dzhengfy) [reply virttest/utils_test/libvirt.py:2716]: > @dzhengfy Just checking the patch for #2988 but I don't think change the return value is a good way as it is widely used.

Yes, we have to change all the tp-libvirt cases which are using it. Anyway if they are working, then it is worth of doing it.

**Review by dzhengfy — APPROVED** [review]
This workaround fix LGTM

**Pull request merged**